CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/rest/RestClientNGTest.java

```diff
@@ -0,0 +1,412 @@
+        String url = "https://testurl.tst/testEndpoint";
+        List<Tuple<String, String>> params = new ArrayList<>();
+        params.add(new Tuple("param1", "value1"));
+        params.add(new Tuple("param2", "value2 with spaces"));
+        URL result = RestClient.generateUrl(url, params);
+        assertEquals(result, new URL(String.format("%s?param1=value1&param2=value2+with+spaces", url)));
+        result = RestClient.generateUrl(url, params, "%20");
+        assertEquals(result, new URL(String.format("%s?param1=value1&param2=value2%%20with%%20spaces", url)));
+    }
+
+    /**
+     * Test of beforeGet method, of class RestClient.
+     */
+    @Test
+    public void testBeforeGet() {
+        System.out.println("testBeforeGet");
+        String url = "";
+        List<Tuple<String, String>> params = null;
+        RestClient instance = new RestClientImpl(0, "", null);
+        instance.beforeGet(url, params);
+        // Method does nothing - designed to be overridden, test really just
+        // checks it can be called
+    }
+
+    /**
+     * Test of get method, of class RestClient.
+     */
+    @Test
+    public void testGetInvalidResponseCode() throws Exception {
+        System.out.println("testGet");
+        String url = "";
+        // Stubbed RestClientImpl uses params to provide class params
+        List<Tuple<String, String>> params = new ArrayList<>();
+        Map<String, List<String>> headerFields = new HashMap<>();
+        headerFields.put("hdr1", new ArrayList<>());
+        RestClientImpl instance = new RestClientImpl(501, "message", headerFields);
+        instance.get(url, params);
+        assertEquals(instance.getResponseCode(), 501);
+        assertEquals(instance.getResponseMessage(), "message");
+        assertEquals(instance.getHeaderFields().size(), headerFields.size());
+        assertTrue(instance.getHeaderFields().containsKey("hdr1"));
+        assertNull(instance.getBytes());
+        // Note: cant see easy way to test valid response code as call to getBody would fail
+    }
+
+    /**
+     * Test of afterGet method, of class RestClient.
+     */
+    @Test
+    public void testAfterGet() {
+        System.out.println("testAfterGet");
+        String url = "";
+        List<Tuple<String, String>> params = null;
+        RestClient instance = new RestClientImpl(0, "", null);
+        instance.afterGet(url, params);
+        // Method does nothing - designed to be overridden, test really just
+        // checks it can be called
+    }
+
+    /**
+     * Test of beforePost method, of class RestClient.
+     */
+    @Test
+    public void testBeforePost() {
+        System.out.println("testBeforePost");
+        String url = "";
+        List<Tuple<String, String>> params = null;
+        RestClient instance = new RestClientImpl(0, "", null);
+        instance.beforePost(url, params);
+        // Method does nothing - designed to be overridden, test really just
+        // checks it can be called
```

Comment:
similar

Reply:
left in as they do add coverage, although as mentioned they don't test 'muxh'